src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+        };
+        for (x = x1; x <= x2; x++) {
+            constexpr auto HIGHESTBIT = (1 << (CURSOR_SIZE_X - 1));
+            uint8_t pixel             = 0;
+            // ScreenMask
+            if (scMask & HIGHESTBIT)
+                pixel = state.backData[dataPos];
+            scMask = static_cast<uint16_t>(scMask << 1);
+            // CursorMask
+            if (cuMask & HIGHESTBIT)
+                pixel = pixel ^ 0x0f;
+            cuMask = static_cast<uint16_t>(cuMask << 1);
+            // Set Pixel
+            INT10_PutPixel(static_cast<uint16_t>(x),
+                           static_cast<uint16_t>(y),
+                           state.page,
+                           pixel);
+            dataPos++;
+        };
+        dataPos = static_cast<uint16_t>(addx2 + dataPos);
+    };
+
+    RestoreVgaRegisters();
+}
+
+// ***************************************************************************
+// DOS driver interface implementation
+// ***************************************************************************
+
+static void UpdateDriverActive()
+{
+    mouse_shared.active_dos = (state.sub_mask != 0);
+    MOUSE_NotifyStateChanged();
+}
+
+static uint8_t GetResetWheel8bit()
+{
+    if (!state.cute_mouse) // wheel requires CuteMouse extensions
+        return 0;
+
+    const int16_t tmp = static_cast<int16_t>(
+            std::clamp(wheel,
+                       static_cast<int16_t>(INT8_MIN),
+                       static_cast<int16_t>(INT8_MAX)));
+    wheel = 0; // reading always int8_t the wheel counter
+
+    // 0xff for -1, 0xfe for -2, etc.
+    return static_cast<uint8_t>((tmp >= 0) ? tmp : (0x100 + tmp));
+}
+
+static uint16_t GetResetWheel16bit()
+{
+    if (!state.cute_mouse) // wheel requires CuteMouse extensions
+        return 0;
+
+    const int16_t tmp = wheel;
+    wheel             = 0; // reading always clears the wheel counter
+
+    // 0xffff for -1, 0xfffe for -2, etc.
+    return static_cast<uint16_t>((tmp >= 0) ? tmp : (0x10000 + tmp));
+}
+
+static void SetMickeyPixelRate(const int16_t ratio_x, const int16_t ratio_y)
+{
+    // According to https://www.stanislavs.org/helppc/int_33-f.html
+    // the values should be non-negative (highest bit not set)
+
+    if ((ratio_x > 0) && (ratio_y > 0)) {
+        constexpr auto X_MICKEY = 8.0f;
+        constexpr auto Y_MICKEY = 8.0f;
+
+        state.mickeys_per_px_x = static_cast<float>(ratio_x) / X_MICKEY;
+        state.mickeys_per_px_y = static_cast<float>(ratio_y) / Y_MICKEY;
+        state.pxs_per_mickey_x = X_MICKEY / static_cast<float>(ratio_x);
+        state.pxs_per_mickey_y = Y_MICKEY / static_cast<float>(ratio_y);
+    }
+}
+
+static void SetDoubleSpeedThreshold(uint16_t threshold)
+{
+    if (threshold)
+        state.dspeed_threshold = threshold;
+    else
+        state.dspeed_threshold = 64; // default value
+}
+
+static void SetSensitivity(uint16_t px, uint16_t py, uint16_t dspeed_threshold)
```

Comment:
Maybe you could avoid mutatings args for consistency, and make these consts

Reply:
Changed.